ReactForte/Application/Webpack/WebpackManifest.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace ReactForte.Application.Webpack;
+
+public class WebpackManifest
+{
+    private readonly WebpackOptions _webpackOptions;
+    private readonly Lazy<IReadOnlyDictionary<string, string>> _localInstance;
+
+    public string ManifestPath => $"{_webpackOptions.OutputPath}/manifest.json";
+
+    public IReadOnlyDictionary<string, string> Instance => _localInstance.Value;
+
+    public WebpackManifest(IOptions<WebpackOptions> webpackOptions, string contentRootPath)
+    {
+        _webpackOptions = webpackOptions.Value;
+        IManifestLoader manifestLoader = _webpackOptions.IsDevServer
+            ? new DevServerLoader(ManifestPath, _webpackOptions.OutputPath)
+            : new LocalLoader(contentRootPath, _webpackOptions.OutputPath, ManifestPath);
+        _localInstance = new Lazy<IReadOnlyDictionary<string, string>>(() => manifestLoader.LoadAsync().Result);
+    }
+
+    private static IEnumerable<string> FilterUrls(IEnumerable<string> urls, IEnumerable<string> pattern,
```

Comment:
Please consider using step-down rule to increase readability -> https://dzone.com/articles/the-stepdown-rule

Reply:
I have moved WebpackManifest to alloy-docker project and changed the implementation to 
`public enum WebpackManifestAssetType
{
    Javascript,
    Style
}

public record WebpackAssets
{
    private readonly IReadOnlyDictionary<string, string> _assets;

    private static readonly IReadOnlyDictionary<WebpackManifestAssetType, string> AssetTypeExtensions =
        new Dictionary<WebpackManifestAssetType, string>
        {
            { WebpackManifestAssetType.Javascript, ".js" },
            { WebpackManifestAssetType.Style, ".css" }
        };

    public WebpackAssets(IReadOnlyDictionary<string, string> assets)
    {
        _assets = assets;
    }

    public WebpackAssets Get(WebpackManifestAssetType assetType)
    {
        return new WebpackAssets(new Dictionary<string, string>(_assets.Where(pair =>
            pair.Value.EndsWith(AssetTypeExtensions[assetType]))));
    }

    public WebpackAssets WithNames(IEnumerable<string> names)
    {
        return new WebpackAssets(new Dictionary<string, string>(_assets.Where(pair => names.Any(pair.Value.Contains))));
    }

    public WebpackAssets Exclude(IEnumerable<string> names)
    {
        return new WebpackAssets(
            new Dictionary<string, string>(_assets.Where(pair => names.Any(pair.Value.Contains) == false)));
    }

    public IList<string> ToList()
    {
        return new List<string>(_assets.Values);
    }
}`